src/pages/mentees.tsx

```diff
@@ -77,6 +78,14 @@ function MenteeTable({ users, refetch }: {
   users: User[],
   refetch: () => void
 }) {
+  const [menteeToYear, setMenteeToYear] = useState<Record<string, string>>({}); 
```

Comment:
Use `Map` instead of `Record`, so a new insert of an existing user id replaces the existing one.